code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -81,7 +81,9 @@
 		EscapeConfinement()
 
 	if(AICanContinue(possible_targets))
-		DestroySurroundings()
+		if(target != null) // fix a runtime error, probably hacky
```

Comment:
if(!QDELETED(target))

Reply:
i dont know what qdeleted does really, but the check is meant mainly for mobs that are spawned without a target, like the lavaland mobs

Reply:
it's the same thing but it catches more cases that are considered to be similar.